File: tests/fixtures/repository-neg/impl.js
// WHEN NOT TO APPLY: data access is already isolated behind a repository
// interface; the service only depends on that abstraction. Nothing to extract.
function makeUserRepo() {
  const users = [{ id: 1, name: "Ada" }];
  return {
    findById(id) {
      return users.find((u) => u.id === id) || null;
    },
  };
}

function greet(repo, id) {
  const u = repo.findById(id);
  return u ? "Hello " + u.name : "unknown";
}

module.exports = { makeUserRepo, greet };

File: tests/fixtures/py-repository-neg/impl.py
# WHEN NOT TO APPLY: data access is already isolated behind a repository
# abstraction; the service only depends on that abstraction. Nothing to extract.


class UserRepo:
    def __init__(self):
        self._users = [{"id": 1, "name": "Ada"}]

    def find_by_id(self, user_id: int):
        for u in self._users:
            if u["id"] == user_id:
                return u
        return None


def greet(repo: UserRepo, user_id: int) -> str:
    u = repo.find_by_id(user_id)
    return "Hello " + u["name"] if u else "unknown"
